api/upload/index.ts

```diff
@@ -12,11 +14,94 @@ const httpTrigger: AzureFunction = async function (
   if (req.body) {
     const { fields, files } = await parseMultipartFormData(req);
     const plot = req.query.plot;
-    // console.log(plot);
+    // console.log(files);
 
-    uploadFiles(files, plot);
-    responseStatusCode = 201;
-    responseMessage = "File uploaded to the cloud successfully";
+    // simple validation here
+    const jsonResults: JSON[] = [];
+    const headers = [
+      "Tag",
+      "Subquadrat",
+      "SpCode",
+      "DBH",
+      "Htmeas",
+      "Codes",
+      "Comments",
+    ];
+
+    // array for collected errors
+    const errors: any = {};
+
+    function createFileEntry(parsedFile: ParsedFile) {
+      if (errors[parsedFile.filename] == undefined) {
+        errors[parsedFile.filename] = {};
+      };
+    }
+
+    for (const parsedFile of files) {
+      // without the transformHeader parameter first header is parsed with quotes
+      
+      const config: ParseConfig = {
+        delimiter: ",",
+        header: true,
+        skipEmptyLines: true,
+        transformHeader: (h) => h.trim(),               
+
+      };
+      const results = parse(parsedFile.bufferFile.toString('utf-8'), config);
+      console.log(results.data);
+
+      // If there is no data, send response immediately?
+      if (!results.data.length) {
+        console.log("No data for upload!");
+        context.res = {
+          status: 400,
+          body: {
+            error: "No data for upload!"
+          },
+        };
+      return
+      }
+
+      results.data.map((csv, index) => {
+        const keys = Object.keys(csv);
+        const currentRow = index + 1;
+
+        if (!keys.every((entry) => headers.includes(entry))) {
+          createFileEntry(parsedFile);
+          console.log("Headers test failed!");
+          errors[parsedFile.filename]["headers"] = "Missing Headers";
+        }
+
+        keys.map((key) => {
+          if (csv[key] === "" || undefined) {
+            createFileEntry(parsedFile);
+            errors[parsedFile.filename][currentRow] = "Missing value in ";
+            console.log(errors[parsedFile.filename][currentRow]);
+          }
+          else if (key === 'DBH' && (parseInt(csv[key]) < 1)) {
+            createFileEntry(parsedFile);
+            errors[parsedFile.filename][currentRow] = "Check the value of DBH";
+            console.log(errors[parsedFile.filename][currentRow]);
+          }
+        });
+      });
+      if (results.errors.length) {
+        createFileEntry(parsedFile);
+        console.log(
+          `Error on row: ${results.errors[0].row}. ${results.errors[0].message}`
+        );
+        errors[parsedFile.filename][results.errors[0].row] = results.errors[0].message;
+      }
+      console.log("Validation errors:", errors);
+    }
+
+    if (!errors.length) {
+      uploadFiles(files, plot);
+      responseStatusCode = 201;
+      responseMessage = "File uploaded to the cloud successfully";
+    } else {
+      // send the response with errors
+    }
```

Comment:
Looks like great progress! I think all that's left to consider this backend piece done is return the error(s) response as a JSON response. The follow up PR will be to have the frontend read that.